gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/domain_service/CreateApiDomainService.java

```diff
@@ -63,10 +65,12 @@ public CreateApiDomainService(
         FlowCrudService flowCrudService,
         NotificationConfigCrudService notificationConfigCrudService,
         ParametersQueryService parametersQueryService,
-        WorkflowCrudService workflowCrudService
+        WorkflowCrudService workflowCrudService,
+        ApiCategoryService apiCategoryService
```

Comment:
When using the onion architecture, services used in the constructors need to be within the onion. So in addition to doing all the work in `ApiCategoryService`, you will need to create a `CategoryService` within the onion architecture that can perform the tasks.

This constructor should then only use the service within the onion. There's a big advantage to testing (no mocks 😱 ) and the code is a bit better in quality.

Reply:
Done